web/src/app/admin/AdminNumberLookup.tsx

```diff
@@ -0,0 +1,187 @@
+import React, { useState } from 'react'
+import { Form } from '../forms'
+import {
+  Button,
+  Card,
+  CardContent,
+  CardActions,
+  Dialog,
+  DialogActions,
+  DialogTitle,
+  Divider,
+  Grid,
+  InputAdornment,
+  List,
+  ListItem,
+  ListItemText,
+  TextField,
+  Tooltip,
+} from '@material-ui/core'
+import gql from 'graphql-tag'
+import { useMutation, useQuery } from 'react-apollo'
+import CopyText from '../util/CopyText'
+import LoadingButton from '../loading/components/LoadingButton'
+import DialogContentError from '../dialogs/components/DialogContentError'
+import { ApolloError } from 'apollo-client'
+
+const carrierInfoMut = gql`
+  mutation($number: String!) {
+    debugCarrierInfo(input: { number: $number }) {
+      name
+      type
+      mobileNetworkCode
+      mobileCountryCode
+    }
+  }
+`
+
+const numInfoQuery = gql`
+  query($number: String!) {
+    phoneNumberInfo(number: $number) {
+      id
+      valid
+      regionCode
+      countryCode
+      formatted
+      error
+    }
+  }
+`
+
+export default function AdminNumberLookup(): JSX.Element {
+  const [number, setNumber] = useState('')
+  const [staleCarrier, setStaleCarrier] = useState(true)
+  const [lastError, setLastError] = useState(null as null | ApolloError)
+
+  const { data: numData } = useQuery(numInfoQuery, {
+    variables: { number: '+' + number },
+    pollInterval: 0,
+    onError: (err) => setLastError(err),
+  })
+
+  const [lookup, { data: carrData, loading: carrLoading }] = useMutation(
+    carrierInfoMut,
+    {
+      variables: { number: '+' + number },
+      onError: (err) => setLastError(err),
+    },
+  )
+
+  function renderListItem(label: string, _text: string): JSX.Element {
+    const text = (_text === undefined ? '' : _text).toString()
```

Comment:
is there a reason we are explicitly checking for undefined? might be best just to check if falsey, i.e. `_text ? _text : ''`

and also we shouldn't need to have a `.toString()` since we have type safety on `_text` coming in.

Reply:
I'm wondering if we need this check at all as since the parameter is not optional, we should always get a string to use. 🤔

Reply:
The issue is the values come from the GraphQL response which is the `any` type -- they can be undefined but the type checking will miss it.

But now that you mention it -- I'll update it to use the schema type and fix the signature accordingly.

Reply:
Sounds good